### PR TITLE
Scanner(unification)

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Unification of methods
* TryScanToken
* ScanTokenFullWidth

* ScanToken_Unified

### CODE REVIEW

**Comment by mmitche** [comment]
This is failing with assertion popups.  @jaredpar Can we avoid popup assertions in the tests?

**Comment by jaredpar** [comment]
@mmitche there are some `Debug.Assert` calls in this code and according to the following xUnit v2 stopped intercepting them hence it will popup a dialog again:

> https://github.com/xunit/resharper-xunit/issues/14

Is this blocking Jenkins or does it dismiss the dialogs after a time?

**Comment by AdamSpeight2008** [comment]
@jaredpar @mmitche The `Debug.Assert` already pre-existed in the Roslyn source.

@jaredpar  it akin to doing a `MessageBox.Show `. How long do I have to wait for it to dismiss automatically?

**Comment by jaredpar** [comment]
@AdamSpeight2008 the `Debug.Assert` calls aren't firing on any of our machines.  It's likely that it's a result of something that you've changed.  Best path forward is to attach a debugger when it shows up and see why it is firing.

**Comment by AdamSpeight2008** [comment]
Still doesn't change that fact that `Debug.Assert` is causing a popup assertion.

**Comment by AnthonyDGreen** [comment]
@AdamSpeight2008,

It wouldn't matter whether the assert was a popup as it is now or whether it was trapped as by xUnit and treated as a unit test failure. The presentation has changed but the end result is that there has been a regression of some pre-existing assumption in compiler. Are all tests passing on your local machine?

**Comment by mmitche** [comment]
The result of this is that the test will hang until the popup is dismissed (this is not automatic).  Aborting the test kills the popup, but that's not ideal.

File: src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj [hunk 1]
```diff
@@ -945,6 +945,7 @@
     <Compile Include="VisualBasicCompilationOptions.vb" />
     <Compile Include="VisualBasicExtensions.vb" />
     <Compile Include="VisualBasicParseOptions.vb" />
+    <Compile Include="_Exts\Char.vb" />
```
Comment (AnthonyDGreen) [inline src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj]: What is _Exts? Did this folder exist beforehand?

**Comment by AnthonyDGreen** [comment]
@mmitche, getting the Asserts to not popup is important but orthogonal. Even if there were no popups the test would still fail and we wouldn't be able to merge in the change. Whatever is causing the assert still has to be addressed.

File: src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj (hunk 1 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj]: Nope, I added it, it's name such to make it obvious is not a blessed namespace.

**Comment by AdamSpeight2008** [comment]
@AnthonyDGreen True. 
At the moment I can't figure it out. #dd49734596 should work as far as I can tell the unified method is correct, for the previous two methods.

**Comment by AnthonyDGreen** [comment]
@AdamSpeight2008,

Thanks for contributing! This PR includes several thousand changes. Many of which are formatting/style changes which aren't focused around the specific change you're making. Because we like to be pretty thorough in our reviews before accepting changes, particularly in areas as performance critical as the scanner the sheer volume of changes in prohibitive. It's not that the changes don't in some cases make the code more readable or more concise, but they make reviewing your core change several orders of magnitude more time consuming. Please revert these edits and constraint this PR to those changes germane to the unification you're suggesting so that the dev owners of that code can remain laser focused on changes with the highest impact and risk.

Additionally, please be more descriptive about the changes you're proposing and their motivation. I understand that there are some methods being unified but it's not immediately clear to us or others what's really being unified, how, and why. Some small code snippets demonstrating the before and after might be beneficial. Having a fairly descriptive overview of the PR is also critical to keeping the PR process as efficient as possible.

**Comment by pharring** [comment]
@AdamSpeight2008 Should this be closed now?

**Comment by AdamSpeight2008** [comment]
@pharring You can close it, when I get it to work I'll submit a new PR

**Comment by jaredpar** [comment]
Please do not submit a new PR. Instead reopen and reuse this one. It helps keep all of the feedback on a single logical change in the same location.

JaredPar from a phone
http://blog.paranoidcoding.com/
________________________________

**Comment by citizenmatt** [comment]
The `Debug.Assert` dialog is default xunit behaviour. I think this should be fixed in xunit, so I've raised an issue: xunit/xunit#382

**Pull request closed without merging**